enhancements/machine-config/alignment-with-image-mode-rhel-for-the-mco.md

```diff
@@ -0,0 +1,205 @@
+    - Why non-goal: This is a valid request and will enhance user experience by increasing customization option.This is a non-goal for this enhancement because this request should be the next step after bootc switch happening. It will be announced and discussed in detail in a follow-up enhancement. 
+* Splitting the RHCOS Image into layers [Discussed in detail in https://github.com/openshift/enhancements/pull/1637]
+
+## Proposal
+
+The efforts needed to fully align the MCO with Image Mode RHEL in 4.19 or 4.20 can be divided into three phases, each building upon the previous one. In the section below, we include a detailed description and dependencies for each phase. 
+
+### Phase 1: On Cluster Layering GA
+
+On-Cluster Layering [Discussed in detail in https://github.com/openshift/enhancements/pull/1515] introduces a seamless image-based configuration flow with details summarized as follows: “perform an on-cluster build (OCB) to combine the base OS image shipped with OpenShift, any admin-provided custom content, and the MachineConfig content managed by the MCO. This image is then deployed to all of the cluster nodes using rpm-ostree (and later, bootc) to facilitate a singular, transactional, update”
+
+On-Cluster Layering is already shipped as a tech preview version in 4.16. Currently, we are targeting 4.18 for its GA. In this phase, we are aiming that, by the time On Cluster Layering GA, it will support most day-2 configurations specified by machine config and ship the change as a container image layer, which include osImageURL specification (DONE), extensions and kernel changes (PLANNED) and file writing to mutable local file system directories (DONE). Another important GA criteria for On Cluster Layering and Phase 1 is that, currently, On Cluster Layering is only a day-2 configuration tool for Openshift, with the workflow being that you install a cluster and then opt in to it. We need to figure out an On Cluster Layering day-1 configuration story so that users can install a cluster with it.
+
+### Phase 2: Move to Layered Update Path by Default 
+
+After On Clustering GA, we will gradually deprecate the current configuration method of direct file writing to disk and move to layered update path (Phase 1) by default. At the end of this phase, OCP nodes will be configured in a full image-based way and will have a RHEL-bootc layer, a CoreOS layer, an OCP node layer, and day-2 configuration layers added on top built and rolled out by On Cluster Layering discussed in Phase 1. Please note here that the proposal and implementation details of the first three layers (the RHEL-bootc layer, the CoreOS layer and the OCP node layer) are explained in the Splitting the RHCOS Image into layers  enhancement, thus a non-goal for this enhancement. Some of the gaps we observed here are:
```

Comment:
How the node image onto which we layer the MachineConfig bits is built/itself composed of layers should be transparent to this work and is mostly orthogonal (i.e. it could happen before or after).

I.e. I'm not sure it's useful to reference those details here.

```suggestion
After On Clustering GA, we will gradually deprecate the current configuration method of direct file writing to disk and move to layered update path (Phase 1) by default. At the end of this phase, OCP nodes will be configured in a full image-based way and day-2 configuration layers added on top of the node image will be built and rolled out by On Cluster Layering discussed in Phase 1. Some of the gaps we observed here are:
```

Reply:
Fixed